### PR TITLE
implement symbol decimal lookup for erc20

### PR DESCRIPTION
**Pull request by ianrowan**
## Link to Issue
Closes: #7579 

## Description of Changes
- Implements stub to lookup contestToken and resolve decimals and symbol for ETH(automatically) and dynamic lookup for any ERC20

## Test Plan
- pass a contestAddress and ensure correct decimals and symbol of contestToken are returned

### CODE REVIEW

**Review by timolegros — CHANGES_REQUESTED** [review]

**Review by rbennettcw — APPROVED** [review]

File: libs/model/src/contest/Contests.projection.ts [hunk 1]
```diff
@@ -38,9 +39,22 @@ async function updateOrCreateWithAlert(
...
-  );
+  const community = await models.Community.findOne({
+    where: { namespace },
+    include: models.ChainNode,
+  });
+
+  if (!community?.ChainNode?.url) {
+    throw new AppError('Chain Node not found');
+  }
+  console.log('env', process.env.NODE_ENV);
```
Comment (timolegros) [inline libs/model/src/contest/Contests.projection.ts:50]:
```suggestion
```

File: libs/model/src/contest/Contests.projection.ts [hunk 2]
```diff
@@ -38,9 +39,22 @@ async function updateOrCreateWithAlert(
...
+  const { ticker, decimals } =
+    process.env.NODE_ENV === 'test'
+      ? { ticker: 'ETH', decimals: 18 }
+      : await contractHelpers.getTokenAttributes(
+          contest_address,
+          new Web3(community?.ChainNode?.url),
```
Comment (timolegros) [inline libs/model/src/contest/Contests.projection.ts:56]:
```suggestion
          new Web3(community?.ChainNode?.private_url || community?.ChainNode?.url),
```

File: libs/model/src/contest/Contests.projection.ts [hunk 3]
```diff
@@ -38,9 +39,22 @@ async function updateOrCreateWithAlert(
   contest_address: string,
   interval: number,
 ) {
-  const { ticker, decimals } = await contractHelpers.getTokenAttributes(
-    contest_address,
-  );
+  const community = await models.Community.findOne({
+    where: { namespace },
+    include: models.ChainNode,
```
Comment (timolegros) [inline libs/model/src/contest/Contests.projection.ts:44]:
```suggestion
    include: models.ChainNode.scope('withPrivateData'),
```

**Review by timolegros — CHANGES_REQUESTED** [review]

**Comment by ianrowan** [comment]
CC @Rotorsoft do you know what's causing this build to fail after I merged master yesterday(related to projections in logs)?

**Review by timolegros — APPROVED** [review]

**Pull request merged**